x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -392,33 +404,73 @@ protected void doRun() throws Exception {
                                         TimeValue.timeValueNanos(statsCurrentTimeNanosSupplier.getAsLong() - startTimeInNanos).millis()
                                     )
                                 );
+                            }));
+                        }
+                        enqueued = true;
+                    } else {
+                        logger.warn("{} too many files ({}) to warm in cache, skipping file [{}]", shardId, queue.size(), fileName);
+                    }
+                } catch (Exception e) {
+                    logger.warn(
+                        () -> new ParameterizedMessage("unable to open file [{}], cancelling prewarming for shard [{}]", fileName, shardId),
+                        e
+                    );
+                    if (queue.size() > 0) {
+                        try {
+                            ActionRunnable<Void> warmer;
+                            while ((warmer = queue.poll(0L, TimeUnit.MILLISECONDS)) != null) {
+                                warmer.onFailure(e);
```

Comment:
Do we actually need this? If my understanding is correct, we are only ever opening any resources on the worker threads (which haven't yet started here), so why do we need to fail the runnables here here?